packages/commcare/test/index.js

```diff
@@ -97,3 +96,74 @@ describe('SubmitXls', () => {
     expect(formdata.get('create_new_cases')).to.equal('on');
   });
 });
+
+describe('getCases', () => {
+  it('should fetch cases', async () => {
+    testServer
+      .intercept({
+        path: `/a/${domain}/api/v0.5/case`,
+        method: 'GET',
+      })
+      .reply(200, () => {
+        // simulate a return from commcare
+        return {
+          data: [
+            {
+              case_id: '12345',
+              properties: { case_type: 'pregnancy', case_name: 'Jane' },
+            },
+          ],
+        };
+      });
+
+    const state = {
+      configuration: {
+        hostUrl,
+        applicationName: domain,
+        appId: app,
+        username: 'user',
+        password: 'password',
+      },
+    };
+
+    const { data } = await execute(get('case'))(state);
+
+    // The response  should be on state.data
+
+    expect(data.data[0]).to.haveOwnProperty('case_id');
+  });
+
+  it('should fetch a single case', async () => {
+    testServer
+      .intercept({
+        path: `/a/${domain}/api/v0.5/case/12345`,
+        method: 'GET',
+      })
+      .reply(200, () => {
+        // simulate a return from commcare
+        return {
+          data: {
```

Comment:
For a single case, the data we get back from commcare is like this:
```
{
  case_id: 'xxx',
  closed: false,
  closed_by: null,
  date_closed: null,
  date_modified: '2024-04-25T15:47:28.193000Z',
  domain: 'test-20190880',
  id: 'xxx,
  indexed_on: '2024-04-25T15:47:31.009517Z',
  indices: {},
  opened_by: 'xxx',
  properties: { ... },
  resource_uri: '',
  server_date_modified: '2024-04-25T15:47:28.247927Z',
  server_date_opened: '2024-04-25T15:47:28.247927Z',
  user_id: 'xxx',
  xform_ids: [ 'xxx' ]
}
```

To be clear, this `reply` function needs to return exactly the same structure as what commcare returns when you GET on `api/v0.5/case/12345`.

This function is mocking out the actual request to commcare - but our adaptor still thinks its making a GET request and getting real data back.

Happy to talk about this a bit more if you're not familiar with mocking in unit tests!